rust/src/ledger/diff/account.rs

```diff
@@ -83,40 +83,51 @@ pub enum AccountDiffType {
 }
 
 impl AccountDiff {
-    pub fn account_creation_payment_diff(new_account: &PublicKey) -> Self {
+    pub fn account_creation_payment_diff(new_account: &PublicKey) -> Vec<Self> {
         let public_key = new_account.clone();
-        AccountDiff::CreateAccount(PaymentDiff {
-            public_key,
-            update_type: UpdateType::Debit(None),
-            amount: Amount(MAINNET_ACCOUNT_CREATION_FEE.0),
-        })
+        vec![
+            AccountDiff::CreateAccount(PaymentDiff {
+                public_key: PublicKey(
+                    "B62qiburnzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzmp7r7UN6X".to_string(),
+                ),
+                update_type: UpdateType::Credit,
+                amount: Amount(MAINNET_ACCOUNT_CREATION_FEE.0),
+            }),
+            AccountDiff::CreateAccount(PaymentDiff {
+                public_key,
+                update_type: UpdateType::Debit(None),
+                amount: Amount(MAINNET_ACCOUNT_CREATION_FEE.0),
+            }),
+        ]
```

Comment:
This is definitely how I think about this creation fee (it is "burned"), but do we know that this is actually how it is? If it isn't, I'd argue that this is more confusing.

Reply:
This also leads to the question of when the burn address is added to the ledger...